### ISSUE TITLE
Incorrect `:unused-referred-var` config produces unexpected behavior

### ISSUE THREAD

**Issue by simonacca** [clj-kondo/clj-kondo#2148]
**version**

`clj-kondo v2023.07.13`

**platform**

JVM

**editor**

Using calva, see [report](https://github.com/BetterThanTomorrow/calva/issues/2265) of the same issue in the Calva repo.

**problem**

Two problems (please let me know if you'd like two separate issues):

1. clj-kondo doesn't warn in some cases when the `:unused-referred-var` config is incorrect and the lint rule is *not* triggered

```clj-kondo --lint - --config '{:linters {:unused-referred-var {:exclude [clojure.set]}}}' <<< "(ns dude (:require [clojure.set :refer [union]])) (union)"```

In this case, I would expect clj-kondo to warn the user that the value corresponding to the `:exclude` key must be a map (see [docs](https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#unused-referred-var)).

2. clj-kondo produces a hard to understand stacktrace  when the `:unused-referred-var` config is incorrect and the lint rule *is* triggered

```clj-kondo --lint - --config '{:linters {:unused-referred-var {:exclude [clojure.set]}}}' <<< "(ns dude (:require [clojure.set :refer [union]]))```

In this case, I would expect clj-kondo to warn the user that the value corresponding to the `:exclude` key must be a map (see [docs](https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#unused-referred-var)).

<details><summary>Stacktrace</summary>
<p>

```
Unexpected error. Please report an issue.
java.lang.IllegalArgumentException: Don't know how to create ISeq from: clojure.lang.Symbol
        at clojure.lang.RT.seqFrom(RT.java:557)
        at clojure.lang.RT.seq(RT.java:537)
        at clojure.core$seq__5467.invokeStatic(core.clj:139)
        at clojure.core$reduce1.invokeStatic(core.clj:932)
        at clojure.core$set.invokeStatic(core.clj:4114)
        at clojure.core$set.invoke(core.clj:4106)
        at clj_kondo.impl.utils$map_vals$fn__3879$fn__3880.invoke(utils.clj:246)
        at clojure.lang.PersistentVector.kvreduce(PersistentVector.java:357)
        at clojure.core$fn__8525.invokeStatic(core.clj:6908)
        at clojure.core$fn__8525.invoke(core.clj:6888)
        at clojure.core.protocols$fn__8257$G__8252__8266.invoke(protocols.clj:175)
        at clojure.core$reduce_kv.invokeStatic(core.clj:6919)
        at clojure.core$reduce_kv.invoke(core.clj:6910)
        at clj_kondo.impl.utils$reduce_map.invokeStatic(utils.clj:241)
        at clj_kondo.impl.utils$reduce_map.invoke(utils.clj:235)
        at clj_kondo.impl.utils$map_vals.invokeStatic(utils.clj:246)
        at clj_kondo.impl.utils$map_vals.invoke(utils.clj:243)
        at clj_kondo.impl.config$eval4607$delayed_cfg__4608.invoke(config.clj:277)
        at clojure.lang.AFn.applyToHelper(AFn.java:154)
        at clojure.lang.AFn.applyTo(AFn.java:144)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$memoize$fn__6946.doInvoke(core.clj:6388)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clj_kondo.impl.config$eval4607$unused_referred_var_excluded__4610.invoke(config.clj:280)
        at clj_kondo.impl.linters$lint_unused_namespaces_BANG_.invokeStatic(linters.clj:557)
        at clj_kondo.impl.linters$lint_unused_namespaces_BANG_.invoke(linters.clj:511)
        at clj_kondo.core$run_BANG_.invokeStatic(core.clj:219)
        at clj_kondo.core$run_BANG_.invoke(core.clj:56)
        at clj_kondo.main$main.invokeStatic(main.clj:149)
        at clj_kondo.main$main.doInvoke(main.clj:135)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.core$apply.invoke(core.clj:662)
        at clj_kondo.main$_main$fn__16203.invoke(main.clj:186)
        at clj_kondo.main$_main.invokeStatic(main.clj:186)
        at clj_kondo.main$_main.doInvoke(main.clj:184)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.core$apply.invokeStatic(core.clj:667)
        at clojure.main$main_opt.invokeStatic(main.clj:514)
        at clojure.main$main_opt.invoke(main.clj:510)
        at clojure.main$main.invokeStatic(main.clj:664)
        at clojure.main$main.doInvoke(main.clj:616)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.main.main(main.java:40)
```

</p>
</details>
